transcript/src/main/java/com/transcriptanalyzer/transcript/User_Requests_Intents/Service/Tree.java

```diff
@@ -0,0 +1,108 @@
+                if (turn.size() == 2) {
+                    String currIntent = turn.get(1);
+                    // dont create node if it's none
+                    if (!Objects.equals(currIntent, "None")) {
+                        Node newRoot = this.treeRootNode.addSubIntent(currIntent);
+                        // add intent if doesnt exist, increment existing count if exists
+                        this.counts.merge(newRoot.intent, 1, Integer::sum);
+                        this.treeRootNode = newRoot;
+                    }
+                }
+            }
+            // at the end of each transcript bring treeNode back to top
+            this.treeRootNode = start;
+        }
+    }
+
+    // class for each individual intent
+    public class Node {
+        private String intent;
+        // # of number it has appeared in its respective turn
+        private int occurrences;
+        private Node parent;
+        private ArrayList<Node> children;
+
+        public Node(String intent, Node parent){
+            this.intent = intent;
+            this.occurrences = 1;
+            this.parent = parent;
+            this.children = new ArrayList<Node>();
+        }
+
+        public Node addSubIntent(String intent) {
+            // if the intent already exists as a child of the parent node
+            if (this.hasChild(intent) != null) {
+                Node node = this.hasChild(intent);
+                // increment occurrences of this node
+                node.occurrences += 1;
+                return node;
+                // if the intent is not a child of the node
+            } else {
+                // create the new node and add it as a child of the parent node
+                Node newChildNode = new Node(intent, this);
+                this.children.add(newChildNode);
+                return newChildNode;
+            }
+        }
+
+        // return subNode if it matches intent, o.w return None
+        public Node hasChild(String intent){
+            for (Node child : this.children) {
+                if (Objects.equals(intent, child.intent)) {
+                    return child;
+                }
+            }
+            return null;
+        }
+    }
+
+    // get a list of the three nodes with the greatest occurrences sorted in decreasing order
+    public List<String> getTopThree(){
+        // TODO: get the top 3 intents for sets: one set is the 3 most common for the whole tree (higher level),
+        //  one is 3 most common for leafs,
+
+        // compares each value and get the top 3, return key list attributed to these top 3
+        return this.counts.entrySet().stream().sorted(
+                Map.Entry.<String, Integer>comparingByValue().reversed()).limit(3).map(
+                Map.Entry::getKey).collect(Collectors.toList());
+    }
+
+    //martin: create top 3 for leaves
+
+    public int add (int numberA, int numberB){
+        return numberA + numberB;
+    }
```

Comment:
Could probably remove this unless it's being used for testing, but in that case should probably be moved to a test file.